### ISSUE TITLE
[Paper] ARES: Automated Evaluation Framework for RAG (Saad-Falcon et al., 2023)

### ISSUE THREAD

**Issue by shaun0927** [shaun0927/ce-study#72]
## Paper Info
- **Title**: ARES: An Automated Evaluation Framework for Retrieval-Augmented Generation Systems
- **Authors**: Jon Saad-Falcon, Omar Khattab, Christopher Potts, Matei Zaharia (Stanford)
- **Venue**: NAACL 2024
- **Links**: [arXiv](https://arxiv.org/abs/2311.09476) | [Code](https://github.com/stanford-futuredata/ARES) | [Paper PDF](https://aclanthology.org/2024.naacl-long.20.pdf)

## TL;DR
RAG 시스템을 **최소한의 인간 라벨링(150개)만으로** 평가하는 프레임워크. **합성 데이터 생성**으로 경량 LLM Judge를 학습하고, **Prediction-Powered Inference(PPI)**로 통계적 신뢰 구간을 제공. RAGAS보다 정확한 RAG 시스템 순위 매기기가 가능하며, 도메인 일반화 성능이 뛰어남.

## Problem
**Core Problem**: RAG 시스템 평가에는 많은 인간 라벨링이 필요하고, LLM Judge의 예측에는 통계적 보장이 없어 신뢰하기 어렵다.

**Motivation**: 
- RAGAS 등 기존 평가는 reference-free지만 통계적 신뢰 구간 미제공
- LLM Judge는 예측 오류가 있을 수 있음
- 대규모 인간 라벨링은 비용이 높음

**Gap**: 
- 통계적으로 보장된 RAG 평가 프레임워크 부재
- 적은 라벨링으로 신뢰할 수 있는 평가 방법 필요
- 도메인 간 일반화 가능한 평가 필요

### Intuition
시험을 볼 때 AI 채점기가 있다고 상상해보자. 이 채점기가 완벽하지 않아서 틀릴 수 있다. ARES는 두 가지로 이 문제를 해결한다: 1) 먼저 AI 채점기를 잘 훈련시키고(합성 데이터), 2) 소수의 인간 채점 결과로 AI 채점기의 오류를 보정한다(PPI). 결과적으로 "이 RAG 시스템의 정확도는 85% ± 3%입니다"처럼 신뢰 구간까지 제공할 수 있다.

#### Three-Stage Pipeline

```
┌─────────────────────────────────────────────────────────────┐
│                    ARES Pipeline                             │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  Stage 1: Synthetic Data Generation                         │
│  ─────────────────────────────────────────                  │
│  Corpus Passages → LLM → Synthetic (Q, A) pairs            │
│                                                             │
│  "Given passage P, generate question Q and answer A"        │
│                                                             │
│  ┌────────────┐    ┌─────────┐    ┌──────────────┐         │
│  │  Passages  │ →  │   LLM   │ →  │ Synthetic QA │         │
│  │  (corpus)  │    │ (GPT-4) │    │   Dataset    │         │
│  └────────────┘    └─────────┘    └──────────────┘         │
│                                                             │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  Stage 2: LLM Judge Training                                │
│  ─────────────────────────────────────────                  │
│  Fine-tune lightweight classifiers for 3 tasks:            │
│                                                             │
│  ┌──────────────────┐                                      │
│  │ Context Relevance │ ← Is context relevant to query?     │
│  └──────────────────┘                                      │
│  ┌──────────────────┐                                      │
│  │ Answer Faithfulness│ ← Is answer grounded in context?   │
│  └──────────────────┘                                      │
│  ┌──────────────────┐                                      │
│  │ Answer Relevance  │ ← Does answer address the query?    │
│  └──────────────────┘                                      │
│                                                             │
│  Training: Contrastive learning objective                  │
│  Model: DeBERTa-v3-large (lightweight)                     │
│                                                             │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  Stage 3: Prediction-Powered Inference (PPI)               │
│  ─────────────────────────────────────────                  │
│                                                             │
│  ┌─────────────────────────────────────────────────────┐   │
│  │                                                     │   │
│  │   Human Annotations (150+)  +  LLM Predictions     │   │
│  │          (small)                  (large)          │   │
│  │              │                        │             │   │
│  │              └──────────┬─────────────┘             │   │
│  │                         │                           │   │
│  │                         ▼                           │   │
│  │              ┌──────────────────┐                   │   │
│  │              │   PPI Rectifier  │                   │   │
│  │              │  (error correction)│                  │   │
│  │              └────────┬─────────┘                   │   │
│  │                       │                             │   │
│  │                       ▼                             │   │
│  │     Score: 85.2% (95% CI: [82.1%, 88.3%])         │   │
│  │                                                     │   │
│  └─────────────────────────────────────────────────────┘   │
│                                                             │
└─────────────────────────────────────────────────────────────┘
```

#### 3가지 평가 메트릭

| Metric | Question | Input |
|--------|----------|-------|
| **Context Relevance** | 검색된 문맥이 질문과 관련 있나? | Query, Context |
| **Answer Faithfulness** | 답변이 문맥에 근거하나? | Context, Answer |
| **Answer Relevance** | 답변이 질문에 적절한가? | Query, Answer |

#### PPI (Prediction-Powered Inference) 상세

```
기존 방식:
───────────────────────────────────────
LLM Judge 예측만 사용
→ "정확도: 85%" (불확실성 없음)
→ Judge가 틀릴 수 있는데 얼마나?


ARES (PPI) 방식:
───────────────────────────────────────
Step 1: LLM Judge로 대량 예측 (N개)
        predictions = [0.9, 0.8, 0.7, ...]

Step 2: 소량의 인간 라벨 (150개)로 오류율 추정
        human_labels = [1, 1, 0, ...]
        error_rate = estimate_bias(predictions, human_labels)

Step 3: 오류 보정 + 신뢰 구간 계산
        corrected_score = raw_score - error_rate
        confidence_interval = compute_ci(alpha=0.95)

→ "정확도: 85.2% ± 3.1% (95% CI)"
→ 통계적 보장!
```

### Implementation Details
- **합성 데이터 생성**: GPT-3.5/4 사용
- **Judge 모델**: DeBERTa-v3-large (fine-tuned)
- **Human Annotations**: 150-300개 권장
- **Few-shot Examples**: 5개 필요
- **설치**: `pip install ares-ai`

### RAGAS vs ARES 비교 (Ranking Accuracy)

| Dataset | RAGAS | GPT-3.5 Judge | ARES |
|---------|-------|---------------|------|
| KILT-NQ | 0.47 | 0.53 | **0.73** |
| KILT-HotpotQA | 0.40 | 0.47 | **0.67** |
| KILT-FEVER | 0.33 | 0.40 | **0.60** |
| Average | 0.40 | 0.47 | **0.67** |

→ ARES가 RAGAS보다 **67% 더 정확**한 순위 매기기

### PPI의 효과

| Setting | Kendall's τ (w/o PPI) | Kendall's τ (w/ PPI) |
|---------|----------------------|---------------------|
| Context Relevance | 0.58 | **0.65** |
| Answer Faithfulness | 0.62 | **0.71** |
| Answer Relevance | 0.55 | **0.63** |

→ PPI 적용 시 모든 메트릭에서 **10-15% 개선**

### 도메인 일반화

- 다른 도메인 데이터로 학습해도 성능 유지
- Query/Document 타입이 달라도 Judge 정확도 유지

## CE Perspective
**Category**: **Select** (평가/품질 측정)

**Key Insight for CE**:

1. **통계적 보장의 중요성**
   ```
   ┌─────────────────────────────────────────┐
   │     CE 기법 평가 시 ARES 활용           │
   ├─────────────────────────────────────────┤
   │                                         │
   │  기존: "이 RAG가 더 좋다" (주관적)      │
   │                                         │
   │  ARES: "RAG A: 85.2% ± 2.1%            │
   │         RAG B: 82.1% ± 2.3%            │
   │         → A가 통계적으로 유의하게 우수" │
   │                                         │
   │  → 신뢰 구간으로 의사결정 가능!         │
   │                                         │
   └─────────────────────────────────────────┘
   ```

2. **효율적인 평가 데이터 확보**
   - 합성 데이터로 Judge 학습 → 라벨링 비용 절감
   - 150개 인간 라벨만으로 통계적 보장

3. **RAGAS와의 조합**
   ```
   RAGAS: 빠른 개발 단계 평가 (reference-free)
      ↓
   ARES: 프로덕션 배포 전 정밀 평가 (with CI)
   ```

4. **A/B 테스트 지원**
   - 신뢰 구간으로 두 시스템 비교
   - "통계적으로 유의한 차이인가?" 답변 가능

## Key Quotes
> "ARES finetunes lightweight LM judges to assess individual RAG components and utilizes a small set of human-annotated datapoints for prediction-powered inference (PPI)." (p. 1)

> "ARES significantly improves data efficiency during evaluation by only requiring 150 annotated datapoints." (p. 3)

> "PPI allows us to estimate confidence intervals with a selected level of probability." (p. 5)

## Strengths
1. **통계적 신뢰 구간**: LLM 평가에 신뢰 구간 제공
2. **데이터 효율성**: 150개 라벨만으로 신뢰성 있는 평가
3. **도메인 일반화**: 다른 도메인에도 적용 가능
4. **RAGAS 대비 우수**: 더 정확한 RAG 시스템 순위 매기기
5. **오픈소스**: 코드 및 데이터셋 공개

## Weaknesses
1. **추가 라벨링 필요**: RAGAS(0개) vs ARES(150개)
2. **Judge 학습 비용**: Fine-tuning 필요
3. **복잡성**: 3단계 파이프라인으로 설정 복잡
4. **합성 데이터 품질**: LLM 생성 품질에 의존

## Related Issues
- Builds on: #62 (RAGAS - 기본 RAG 평가)
- Related to: #57 (L-Eval - Long Context 평가)
- Compare with: Phase 7 Evaluation 논문들

## Action Items
- [ ] ARES 설치 및 기본 사용법 익히기
- [ ] RAGAS와 ARES 비교 실험
- [ ] 150개 validation set 구성 가이드라인 정리
- [ ] PPI 통계 방법론 심층 이해

### RAGAS vs ARES 비교

```
┌────────────────────────────────────────────────────────────┐
│                RAGAS vs ARES 비교                           │
├────────────────────────────────────────────────────────────┤
│                                                            │
│  특성           │  RAGAS          │  ARES                  │
│  ──────────────────────────────────────────────────────   │
│  인간 라벨      │  0개 (대부분)   │  150개                 │
│  통계적 보장    │  없음           │  95% 신뢰 구간        │
│  Judge         │  GPT-4 직접     │  Fine-tuned 경량 모델  │
│  순위 정확도   │  0.40 (avg τ)  │  0.67 (avg τ)         │
│  복잡도        │  낮음           │  중간                  │
│  비용          │  API 비용만     │  학습 + 라벨링 비용   │
│  사용 시점     │  개발 단계      │  배포 전 검증         │
│                                                            │
└────────────────────────────────────────────────────────────┘
```

### 사용 시나리오

```
개발 초기:
  → RAGAS로 빠른 피드백

프로덕션 전:
  → ARES로 정밀 평가 + 신뢰 구간

A/B 테스트:
  → ARES의 CI로 통계적 유의성 검증

모델 비교:
  → ARES의 ranking 정확도 활용
```

### PPI 핵심 아이디어

```python
# Simplified PPI concept
def ares_score_with_ppi(llm_predictions, human_labels):
    """
    llm_predictions: 대량의 LLM Judge 예측 (N개)
    human_labels: 소량의 인간 라벨 (150개)
    """
    # Step 1: LLM 예측의 편향 추정
    labeled_preds = llm_predictions[:150]
    bias = mean(labeled_preds - human_labels)
    
    # Step 2: 보정된 점수 계산
    corrected_score = mean(llm_predictions) - bias
    
    # Step 3: 신뢰 구간 계산
    ci = compute_confidence_interval(
        llm_predictions, 
        human_labels, 
        alpha=0.95
    )
    
    return corrected_score, ci

# 결과: 85.2%, (82.1%, 88.3%)
```

---
**Reading Status**: [ ] 1st pass [x] Deep read [ ] Implemented
**Confidence**: High
**CE Relevance**: ⭐⭐⭐⭐⭐ (통계적으로 보장된 RAG 평가)
